force_model.py

```diff
@@ -0,0 +1,332 @@
+#need to fix above # fixed
+
+##second section, Define constants
+alpha = 0.93; # Sentman model's accommodation coefficient
+
+#getarenrm function
+def getareanrm():
+    #data file with macromodel plate areas (m^2) and plate normal vectors in spacecraft body-fixed coordinates data refers to             #Starlink v1.0 and body-fixed coordinate system as shown in Starlink_geometry_20200331.jpg reflectivities and materials are           #guesses for now, satellite bus is assumed completely flat... need to revisit this at some point
+    Au = 196.96655
+    #SiO2 = 60.0843
+    SiO2 = 20.03
+    Al = 26.981538
+
+    #arrays for returns
+    A = np.zeros(6)
+    plt_M = np.zeros(6)
+    bwspec = np.zeros(6)
+    bwdiff = np.zeros(6)
+    nrm_sbf = np.zeros((3,6))
+
+
+    #****SOLAR PANELS: Front panel -Y (Shark Fin mode) or -Z (Open Book mode, sada_pos=pi/2; default setting used here, to be changed     #outside of this routine)
+    nrm_sbf[:,0] = [0, 0, -1] 
+    A[0] = 8.05*2.77
+    bwspec[0] = 0.05
+    bwdiff[0] = 0.3
+    plt_M[0] = SiO2
+
+    #****SOLAR PANELS: Back panel +Y (Shark Fin mode) or +Z (Open Book mode, sada_pos=0; default setting used here, to be changed         #outside of this routine)
+    nrm_sbf[:,1] = [0, 0, 1] 
+    A[1] = 8.05*2.77
+    bwspec[1] = 0.2
+    bwdiff[1] = 0.4
+    plt_M[1] = Al
+    
+    #****BUS: Zenith-facing -Z
+    nrm_sbf[:,2] = [0, 0, 1] 
+    A[2] = 2.82*1.27
+    bwspec[2] = 0.2
+    bwdiff[2] = 0.4
+    plt_M[2] = Al 
+
+    #***BUS: Nadir-facing +Z
+    nrm_sbf[:,3] = [0, 0, -1] 
+    A[3] = 2.82*1.27
+    bwspec[3] = 0.2
+    bwdiff[3] = 0.4
+    plt_M[3] = Al    
+
+    #****VISORS: Satellite-facing +Y
+    nrm_sbf[:,4] = [0, 1, 0] 
+    A[4] = .681+.426
+    bwspec[4] = 0.2
+    bwdiff[4] = 0.4
+    plt_M[4] = Al  
+
+    #****VISORS: Outward-facing -Y
+    nrm_sbf[:,5] = [0, -1, 0] 
+    A[5] = .681+.426
+    bwspec[5] = 0.2
+    bwdiff[5] = 0.4
+    plt_M[5] = Al     
+
+    for i in range(len(nrm_sbf)):
+        nrm_sbf[:,i] = nrm_sbf[:,i] / np.linalg.norm(nrm_sbf[:,i])
+
+
+    return  [nrm_sbf,A,plt_M,bwspec,bwdiff]
+
+##third section, Read Satellite geometry and properties
+satprop = []
+
+[nrm_sbf,area,plt_M,bwspec,bwdiff] = getareanrm()
+scmass = 260 #Starlink v1.0 mass (kg) this is from wikipedia(!)
+
+#according to Eric, may or may not need satprop
+satprop.append(nrm_sbf)
+satprop.append(area)
+satprop.append(plt_M)
+satprop.append(bwspec)
+satprop.append(bwdiff)
+satprop.append(scmass)
+
+# load matfile and access names for use 
+att_model = scipy.io.loadmat('attitude_model.mat')
+att_model.keys()
+mean_slonsbf = att_model['mean_slonsbf']
+# mean_slonsbf[30,30]
+mean_ssada = att_model['mean_ssada']
+# mean_ssada[30,30]
+
+#2nd function ecef_llh
+
+def ecef_llh(r_ecef,converge):
+    #lat, lon in degrees
+    #height in km
+    a = 6378.1370; # km
+    b = 6356.75231425; # km
+    e = math.sqrt((a**2 - b**2)/a**2); # km
+
+    x = r_ecef[:,0] # km, first value

```

Comment:
These are swapped from the Matlab code it looks like.
```suggestion
    x = r_ecef[0, :] # km, first value
```